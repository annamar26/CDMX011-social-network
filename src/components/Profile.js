/* eslint-disable max-len */
/* eslint-disable no-nested-ternary */
import { fbFunctions } from '../firebaseClient.js';
import { onNavigate } from '../routes.js';

export const Profile = () => {
  const container = document.createElement('div');
  const html = `<nav>
    <div>
    <img class=' logo logo1' src='./icons&img/logo.png'> </div>
    <div id='profile' class='icons img-profile'>
   <img class='logout'src='./icons&img/logout.svg'>
    </div></nav>
  
    <div class='profile-conteiner'>
    <div class='profile'>
    <h2 class='instruccion'>Configura tu perfil</h2>
  <img id='profileImage' src=''>
  <form class='profile-form'> 
 
  <label for='Nombre'>Nombre del perfil</label>
  <input id='Nombre' value='' type=text placeholder='Pablito Perez'>
  <label for='Email'>Email</label>
  <input id='Email' value='' type=text >
  <input id='photo' value='Selecciona tu foto de perfil'  type=file acept='image/*'>
 
  <button id='publicar' class='button' type=submit>Guardar Cambios</button></form>
 </div>
   <div id='postUserCointainer' class='profile-timeline posts-container'></div>
   <div id='footer-conteiner'>
  <footer id='footer'>
  <img class='icon' id='createPost' src='./icons&img/createicon.svg'>  </div>
    </footer>
    </div>`;

  container.innerHTML = html;
  fbFunctions.getCurrentUserPosts();
  fbFunctions.comprobar();
  fbFunctions.setDataProfile();

  const logout = container.querySelector('.logout');
  logout.addEventListener('click', () => {
    fbFunctions.userLogout().then(() => {
      console.log('logout exitoso');
    });
  });

  container.querySelector('#createPost').addEventListener('click', (e) => {
    e.preventDefault();
    onNavigate('/createpost');
  });

  const data = container.querySelector('.profile-form');
  data.addEventListener('submit', (e) => {
    e.preventDefault();
    const name = document.getElementById('Nombre').value;
    const email = document.getElementById('Email').value;
    const photo = document.getElementById('photo').files[0];

    fbFunctions.updateStringsProfile(name, email)
      .then(() => {
        console.log('Perfil editado');
      })
      .catch((error) => {
        console.error('Error writing document: ', error);
      });

    fbFunctions.pushNewPhoto(photo, photo.name)
      .then(() => {
        console.log('Perfil editado');
        fbFunctions.pullNewPhoto(photo.name).then((url) => {
          fbFunctions.updatePhotoProfile(url).then(() => {
            fbFunctions.updateUserDoc(name, email, url);
            onNavigate('/profile');
          });
        });
      })
      .catch((error) => {
        console.error('Error writing document: ', error);
      });
  });

  return container;
};
