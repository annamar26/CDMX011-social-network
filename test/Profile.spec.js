/**
 * @jest-environment jsdom
 */
import { Profile } from '../src/components/Profile.js';
import { render } from '../src/utils.js';

describe('Profile', () => {
  document.body.innerHTML = '<div id="root"></div>';
  const rootDiv = document.getElementById('root');
  it('should render Profile page', () => {
    render(rootDiv, Profile());

    expect(rootDiv).toMatchSnapshot();
  });
});
