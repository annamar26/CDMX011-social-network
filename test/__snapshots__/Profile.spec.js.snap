// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should render Profile page 1`] = `"undefined"`;
